mojap_metadata/metadata/metadata.py

```diff
@@ -1,22 +1,43 @@
 import json
-from warnings import warn
+import yaml
+from copy import deepcopy
 
 import jsonschema
 
 
 class MetadataProperty:
     def __set_name__(self, owner, name) -> None:
-        self.name = f"_{name}"
+        self.name = name
 
     def __get__(self, obj, type=None) -> object:
-        return getattr(obj, self.name)
+        return obj._data.get(self.name)
 
     def __set__(self, obj, value) -> None:
-        setattr(obj, self.name, value)
+        obj._data[self.name] = value
         obj.validate()
 
 
 class Metadata:
+
+    @classmethod
+    def from_dict(cls, d: dict) -> object:
+        m = cls()
+        m._data = deepcopy(d)
+        m.validate()
+        return m
+
+    @classmethod
+    def from_json(cls, filename, **kwargs) -> object:
+        with open(filename, "r") as f:
+            obj = json.load(f, **kwargs)
+            return cls.from_dict(obj)
+
+    @classmethod
+    def from_yaml(cls, filename, **kwargs) -> object:
+        with open(filename, "r") as f:
+            obj = yaml.safe_load(f, **kwargs)
+            return cls.from_dict(obj)
+
```

Comment:
Making these class methods I think makes more sense. The class method returns the object instance of that class.